iceoryx2-ffi/cxx/tests/src/service_request_response_tests.cpp

```diff
@@ -0,0 +1,1625 @@
+}
+
+TYPED_TEST(ServiceRequestResponseTest, open_or_create_service_does_exist) {
+    constexpr ServiceType SERVICE_TYPE = TestFixture::TYPE;
+
+    const auto service_name = iox2_testing::generate_service_name();
+
+    ASSERT_FALSE(
+        Service<SERVICE_TYPE>::does_exist(service_name, Config::global_config(), MessagingPattern::RequestResponse)
+            .expect(""));
+
+    auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
+
+    {
+        auto sut = iox::optional<PortFactoryRequestResponse<SERVICE_TYPE, uint64_t, void, uint64_t, void>>(
+            node.service_builder(service_name)
+                .template request_response<uint64_t, uint64_t>()
+                .open_or_create()
+                .expect(""));
+
+        ASSERT_TRUE(
+            Service<SERVICE_TYPE>::does_exist(service_name, Config::global_config(), MessagingPattern::RequestResponse)
+                .expect(""));
+
+        auto sut_2 = iox::optional<PortFactoryRequestResponse<SERVICE_TYPE, uint64_t, void, uint64_t, void>>(
+            node.service_builder(service_name)
+                .template request_response<uint64_t, uint64_t>()
+                .open_or_create()
+                .expect(""));
+
+        ASSERT_TRUE(
+            Service<SERVICE_TYPE>::does_exist(service_name, Config::global_config(), MessagingPattern::RequestResponse)
+                .expect(""));
+
+        sut.reset();
+
+        ASSERT_TRUE(
+            Service<SERVICE_TYPE>::does_exist(service_name, Config::global_config(), MessagingPattern::RequestResponse)
+                .expect(""));
+
+        sut_2.reset();
+    }
+
+    ASSERT_FALSE(
+        Service<SERVICE_TYPE>::does_exist(service_name, Config::global_config(), MessagingPattern::RequestResponse)
+            .expect(""));
+}
+
+TYPED_TEST(ServiceRequestResponseTest, opening_non_existing_service_fails) {
+    constexpr ServiceType SERVICE_TYPE = TestFixture::TYPE;
+
+    const auto service_name = iox2_testing::generate_service_name();
+
+    auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
+    auto sut = node.service_builder(service_name).template request_response<uint64_t, uint64_t>().open();
+    ASSERT_TRUE(sut.has_error());
+    ASSERT_THAT(sut.error(), Eq(RequestResponseOpenError::DoesNotExist));
+}
+
+TYPED_TEST(ServiceRequestResponseTest, opening_existing_service_works) {
+    constexpr ServiceType SERVICE_TYPE = TestFixture::TYPE;
+
+    const auto service_name = iox2_testing::generate_service_name();
+
+    auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
+    auto sut_create =
+        node.service_builder(service_name).template request_response<uint64_t, uint64_t>().create().expect("");
+    auto sut = node.service_builder(service_name).template request_response<uint64_t, uint64_t>().open();
+    ASSERT_TRUE(sut.has_value());
+}
+
+TYPED_TEST(ServiceRequestResponseTest, opening_existing_service_with_wrong_payload_type_fails) {
```

Comment:
Can you please test this for:
* wrong request payload type
* wrong request user header type
* wrong response payload type
* wrong response user header type

Reply:
Wrong request payload type and response payload type are already checked in this test; I added an additional test for the wrong request/response user header type.